HW_4_Task_1/HW_4_Task_1/Tree.cs

```diff
@@ -0,0 +1,67 @@
+﻿namespace HW_4_Task_1
+{
+    /// <summary>
+    /// Data structure used to store the arithmetic tree.
+    /// </summary>
+    public class Tree
+    {
+        private INode root;
+        private int cursor;
+
+        /// <summary>
+        /// Builds an arithmetic tree.
+        /// </summary>
+        public void Build(string[] expression)
+        {
+            root = BuildNode(expression, ref cursor);
+        }
+
+        private INode BuildNode(string[] expression, ref int cursor)
+        {
+            
+            while (!IsOperator(expression[cursor]))
+            {   
+                int value;
+                if (int.TryParse(expression[cursor], out value))
+                {
+                    cursor++;
+                    return new Operand(value);
+                }
+                if (!IsCorrectElement(expression[cursor]))
+                {
+                    throw new CorrectExpressionException();
+                }
+                cursor++;
+            }
+            
+            if (IsOperator(expression[cursor]))
+            {
+                cursor++;
+                return new Operator(expression[cursor - 1], BuildNode(expression, ref cursor), BuildNode(expression, ref cursor));
+            }
+            else
+            {
+                return null;
+            }
```

Comment:
```suggestion
            return null;
```
Всё равно если условие if истинное, управление сюда не дойдёт